### ISSUE TITLE
[Wallet Connect] Failed to buy collectible on OpenSea when swap is involved

### ISSUE THREAD

**Issue by alexjba** [status-im/status-desktop#16117]
# Bug Report

Used dapp: OpenSea

Buying this collectible on optimism:
https://opensea.io/assets/optimism/0x0deaac29d8a3d4ebbaaa3ecd3cc97c9def00f720/5681

at the end (after offering and converting eth to reth) tx failed with:

`status-go - hashTypedDataV4 failed   topics="wallet-connect-service" tid=34581919 file=service.nim:208 err="there is extra data provided in the message (11 < 12)"`

`ERR 2024-08-13 23:21:25.235+02:00 safeSignTypedData failed:  topics="wallet-connect-controller" tid=34581919 file=controller.nim:162 msg="hashTypedData failed"`

It could be up to the dapp! We have to check.

Issue identified by @saledjenic in https://docs.google.com/spreadsheets/d/1QkV08jm0ECV01x3yWTLnVGCjTluKQ-IBDJXsHjNeAQg/edit?gid=0#gid=0

**Comment by virginiabalducci** [comment]
I was able to purchase a collectible on Open Sea via Optimism Network

https://github.com/user-attachments/assets/f3a9493f-ad54-4a3e-9a89-3aa6d4edaee1
Attaching console log for reference
[app_20240905_105554.log](https://github.com/user-attachments/files/16898582/app_20240905_105554.log)

**Comment by saledjenic** [comment]
I've tested this in the latest master and it works.

Here is the tx that I made:
https://optimistic.etherscan.io/tx/0x166c112e4214eb0e6f9ea9cf55bd4f24dd082b27d13730a0e17d37a2bbeea997

For buying this collectible on optimism:
https://opensea.io/assets/optimism/0x0deaac29d8a3d4ebbaaa3ecd3cc97c9def00f720/1314

----

What I've noticed as an issue that possibly caused confusion is, if user has Status Browser Connect active and wants to connect to the dapp via WalletConnect, on the dapp's page refresh popup for connecting via BC will be offered first. Using it mainnet chain is always displayed for connecting desired account, later like in this case if the user wants to purchase collectible on other that the mainnet chain won't be able, cause the tx will fail. But if user rejects that popup and proceeds with connecting the dapp via WalletConnect all is good. That's the path I've tested.

What we need to determine in this scenario is, what behaviour we want if BC plugin is installed and user wants to use WC instead of BC? @alexjba 

Anyhow, I would close this issue since that part (browser connect for sure, but wallet connect will be touched as well) will be improved very soon. @virginiabalducci @anastasiyaig wdyt?
